app/models/user.rb

```diff
@@ -694,4 +694,8 @@ def password_matches_confirmation
   def strip_payment_pointer
     self.payment_pointer = payment_pointer.strip if payment_pointer
   end
+
+  def confirmation_required?
+    SiteConfig.smtp_enabled?
+  end
```

Comment:
Oh I just noticed a problem with this code. If smtp does is enabled, all the previously created user will need to confirm their account. This is a problem and I'm trying to come up with a solution.

Reply:
Users have `confirmed_at` and `confirmed?`, can we use that?

```ruby
def confirmation_required?
  !confirmed? && SiteConfig.smtp_enabled?
end
```

Reply:
we could but that logic still doesn't account for the scenario I mentioned though. `confirmation_required?` doesn't confirm the user; they remained unconfirmed and will be asked to re-confirmed if the SMTP is ever enabled in the future.

Reply:
I'm starting to believe that we should require people to setup a SMTP server before they start the app, as Discourse does.